code/modules/antagonists/paradox_clone/paradox_clone.dm

```diff
@@ -0,0 +1,96 @@
+/datum/antagonist/paradox_clone
+	name = "Paradox Clone"
+	roundend_category = "Paradox Clone"
+	job_rank = ROLE_PARADOX_CLONE
+	antagpanel_category = "ParadoxClone"
+	antag_hud_name = "paradox_clone"
+	show_to_ghosts = TRUE
+	preview_outfit = /datum/outfit/paradox_clone
+
+	///Weakref to the mind of the original, the clone's target.
+	var/datum/weakref/original_ref
+
+/datum/antagonist/paradox_clone/get_preview_icon()
+	var/icon/final_icon = render_preview_outfit(preview_outfit)
+
+	final_icon.Blend(make_background_clone_icon(preview_outfit), ICON_UNDERLAY, -8, 0)
+	final_icon.Scale(64, 64)
+
+	return finish_preview_icon(final_icon)
+
+/datum/antagonist/paradox_clone/proc/make_background_clone_icon(datum/outfit/clone_fit)
+	var/mob/living/carbon/human/dummy/consistent/clone = new
+
+	var/icon/clone_icon = render_preview_outfit(clone_fit, clone)
+	clone_icon.ChangeOpacity(0.5)
+	qdel(clone)
+
+	return clone_icon
+
+/datum/antagonist/paradox_clone/on_gain()
+	owner.special_role = ROLE_PARADOX_CLONE
+	return ..()
+
+/datum/antagonist/paradox_clone/on_removal()
+	//don't null it if we got a different one added on top, somehow.
+	if(owner.special_role == ROLE_PARADOX_CLONE)
+		owner.special_role = null
+	original_ref = null
+	return ..()
+
+/datum/antagonist/paradox_clone/Destroy()
+	original_ref = null
+	return ..()
+
+/datum/antagonist/paradox_clone/proc/setup_clone()
+	var/datum/mind/original_mind = original_ref?.resolve()
+
+	var/datum/objective/assassinate/paradox_clone/kill = new
+	kill.owner = owner
+	kill.target = original_mind
+	kill.update_explanation_text()
+	objectives += kill
+
+	owner.set_assigned_role(SSjob.GetJobType(/datum/job/paradox_clone))
+
+	//clone doesnt show up on message lists
+	var/obj/item/modular_computer/pda/messenger = locate() in owner.current
+	if(messenger)
+		var/datum/computer_file/program/messenger/message_app = locate() in messenger.stored_files
```

Comment:
You gotta check the HDD instead, and look for messenger app there. This is because yogstation is behind in the times.